modules/carto/src/layers/h3-tileset-2d.ts

```diff
@@ -66,12 +66,20 @@ function tileToBoundingBox(index: string): GeoBoundingBox {
 // similar
 // Relative scale factor (0 = no biasing, 2 = a few hexagons cover view)
 const BIAS = 2;
-export function getHexagonResolution(viewport): number {
+export function getHexagonResolution(viewport: {
+  zoom: number;
+  latitude: number;
+  tileSize: number;
+}): number {
   const hexagonScaleFactor = (2 / 3) * viewport.zoom;
   const latitudeScaleFactor = Math.log(1 / Math.cos((Math.PI * viewport.latitude) / 180));
+  const tileSizeScaleFactor = Math.log2(512 / viewport.tileSize);
```

Comment:
I've tested this visually for 512px and 1024px sizes, and the unit test results look plausible at 2048px. But my understanding of the math above is fuzzy, so other sanity-checks here would be welcome. 

Do you know if there's an existing way to debug/preview the selected H3 tiles, similar to https://labs.mapbox.com/what-the-tile/ ? And does that seem worthwhile to set up, if not?

Reply:
I have this old test app which you could likely adapt: https://jsfiddle.net/felixp/pcrydtLg/239/

Reply:
Your current maths will result in a step in `tileSize` (e.g. 512->1024) corresponding to a step in h3 resolution, which is incorrect. Instead, I think you want:

```ts
// Number of zoom levels that tileSize differs from deck's internal 512 tile size
const zoomOffset = Math.log2(tileSize / 512);
const hexagonScaleFactor = (2 / 3) * (viewport.zoom - zoomOffset);
```

Reply:
Hm... with either the current code in the PR, or your suggested changes, the unit tests are happy and the number of queries is generally reduced. So far so good.

As far as which is a correct or incorrect — I think the choice would be subjective, if we didn't also need to sync to `aggregationResLevel`. But we've consistently incremented `aggregationResLevel` by one, and I think that implies we need a consistent -1 step in h3 resolution? The 2/3 scaling and Math.floor() here mean that if we're not careful, for some zoom levels we may get the same h3 resolution for 1024px tile size as we would have for 512px tile size.

Adapting the code from your JSFiddle, I've pushed a demo here:

https://github.com/visgl/deck.gl/compare/donmccurdy/feat-h3-tileset-tilesize...donmccurdy/demo-h3-tileset-tilesize

In the screen captures below, the `#ff0000` outlines represent hexagons corresponding to 512px tileSize, and the dashed `#00ff00` outlines correspond to 1024px tileSize. Local environment in top half of screen capture, production in lower half.

### A

As proposed by comment above, note that the choice of h3 resolution coincides at some map zoom levels, and aggregation resolution level changes at different zoom levels than before.

```javascript
const zoomOffset = Math.log2(512 / tileSize);
const hexagonScaleFactor = (2 / 3) * (viewport.zoom + zoomOffset);
return Math.max(0, Math.floor(hexagonScaleFactor + latitudeScaleFactor - BIAS));
```

[h3_steps_offset.webm](https://github.com/visgl/deck.gl/assets/1848368/5231610b-02ce-4526-a19e-ad9df888eabf)


### B

Equivalent to the current PR, note that the aggregationResLevel stays in sync with the 

```javascript
const zoomOffset = Math.log2(512 / tileSize);
const hexagonScaleFactor = (2 / 3) * viewport.zoom;
const latitudeScaleFactor = Math.log(1 / Math.cos((Math.PI * viewport.latitude) / 180));
return Math.max(0, Math.floor(hexagonScaleFactor + latitudeScaleFactor + zoomOffset - BIAS));
```

[h3_steps_sync.webm](https://github.com/visgl/deck.gl/assets/1848368/94275205-bdda-4617-80f1-c2f6e51c80dc)

Reply:
Thanks for making the comparison. The difference mathematically between A and B is just the `2 / 3` scaling factor applied to `zoomOffset`. Without this I don't think the actual on-screen size of the tiles will be correct. For tile sizes near to 512 the difference is small, but as we get further away from this (e.g. `tileSize` 128 will drop the h3 resolution by 2, and a h3 hexagon at resolution 4 is not a quarter of the size of one at 6)

In your screen captures what is the `tileSize` used for the requested tiles? I'm guessing it is `1024`, because for `512` the maths for A and B is equivalent (`zoomOffset = 0`).

In order to make the switch in the future from the default `tileSize` of `512` to `1024` in such as way as to keep the displayed cells the same across viewport zoom levels we should adjust the `BIAS` by `1 / 3` to compensate. Could you try and see if you can match the new behavior with the existing this way?

Reply:
You're correct, of course. :) 

Adjusting BIAS ...

```javascript
const BIAS = 2;

...

BIAS += zoomOffset / 3;
```

... and testing `tileSize = 256 | 512 | 1024`, this keeps displayed cells at the same size as  before.